<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SpikeGenDlg</class>
 <widget class="QDialog" name="SpikeGenDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>774</width>
    <height>679</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Spike generator</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="4" column="1" colspan="2">
    <widget class="QPushButton" name="btnLoadST">
     <property name="text">
      <string>Load</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QCheckBox" name="Loop">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="text">
      <string>Loop burst sequence</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="0" rowspan="2" colspan="5">
    <widget class="QWidget" name="widget" native="true">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="minimumSize">
      <size>
       <width>341</width>
       <height>241</height>
      </size>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0" colspan="5">
       <widget class="QLineEdit" name="leLabel">
        <property name="placeholderText">
         <string>Label...</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QLabel" name="titleLabel">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="font">
         <font>
          <pointsize>10</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string/>
        </property>
        <property name="frameShape">
         <enum>QFrame::Panel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <property name="lineWidth">
         <number>1</number>
        </property>
        <property name="midLineWidth">
         <number>0</number>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QLabel" name="SpikesL">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Spikes</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4" rowspan="7">
       <widget class="Line" name="line_3">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QLabel" name="BurstDetectionL">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Burst detection</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" rowspan="2" colspan="3">
       <widget class="QComboBox" name="SGLUTableCombo">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <item>
         <property name="text">
          <string>Direct Functions</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Lookup Tables</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="5">
       <widget class="QComboBox" name="BurstDetectionCombo">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <item>
         <property name="text">
          <string>Off</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>low-&gt;high detection</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>high-&gt;low detection</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="5" rowspan="5">
       <widget class="QWidget" name="bdGroupW" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="NUnderL">
           <property name="text">
            <string>tUnder</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="tUnder">
           <property name="maximum">
            <double>100000.000000000000000</double>
           </property>
           <property name="value">
            <double>2.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="WidthU_2">
           <property name="text">
            <string>ms</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1" colspan="2">
          <widget class="QCheckBox" name="tUContiguous">
           <property name="text">
            <string>Contiguous</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="NOverL">
           <property name="text">
            <string>tOver</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="tOver">
           <property name="maximum">
            <double>100000.000000000000000</double>
           </property>
           <property name="value">
            <double>2.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLabel" name="WidthU_3">
           <property name="text">
            <string>ms</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1" colspan="2">
          <widget class="QCheckBox" name="tOContiguous">
           <property name="text">
            <string>Contiguous</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0" colspan="3">
          <widget class="QCheckBox" name="bdStrict">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Strictly contiguous</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="VSpikeL">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>VSpike</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="2">
       <widget class="QDoubleSpinBox" name="VSpike">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <double>-1000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
        <property name="value">
         <double>80.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QLabel" name="VSpikeU">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>mV</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="WidthL">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Width</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1" colspan="2">
       <widget class="QDoubleSpinBox" name="Width">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="value">
         <double>2.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QLabel" name="WidthU">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>ms</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="VRestL">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>VRest</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1" colspan="2">
       <widget class="QDoubleSpinBox" name="VRest">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <double>-1000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
        <property name="value">
         <double>-50.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QLabel" name="VRestU">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>mV</string>
        </property>
       </widget>
      </item>
      <item row="7" column="2">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>65</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
     <zorder>bdGroupW</zorder>
     <zorder>SpikesL</zorder>
     <zorder>VRestL</zorder>
     <zorder>VRestU</zorder>
     <zorder>VSpikeL</zorder>
     <zorder>VSpikeU</zorder>
     <zorder>titleLabel</zorder>
     <zorder>line_3</zorder>
     <zorder>WidthL</zorder>
     <zorder>BurstDetectionCombo</zorder>
     <zorder>WidthU</zorder>
     <zorder>BurstDetectionL</zorder>
     <zorder>SGLUTableCombo</zorder>
     <zorder>VSpike</zorder>
     <zorder>Width</zorder>
     <zorder>VRest</zorder>
     <zorder>leLabel</zorder>
     <zorder>verticalSpacer</zorder>
    </widget>
   </item>
   <item row="0" column="5">
    <widget class="QLabel" name="SpikesL_2">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>21</height>
      </size>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Instances</string>
     </property>
    </widget>
   </item>
   <item row="3" column="4">
    <widget class="QLabel" name="PeriodU">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="text">
      <string>ms</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>21</height>
      </size>
     </property>
     <property name="text">
      <string>Spike times in ms, one burst per row:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="2" colspan="2">
    <widget class="QDoubleSpinBox" name="Period">
     <property name="maximum">
      <double>6000000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>100.000000000000000</double>
     </property>
     <property name="value">
      <double>500.000000000000000</double>
     </property>
    </widget>
   </item>
   <item row="6" column="6">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Period</string>
     </property>
    </widget>
   </item>
   <item row="4" column="3" colspan="2">
    <widget class="QPushButton" name="btnSaveST">
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
   </item>
   <item row="1" column="5" rowspan="4" colspan="2">
    <widget class="QTableWidget" name="instTable">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="columnCount">
      <number>7</number>
     </property>
     <attribute name="horizontalHeaderMinimumSectionSize">
      <number>10</number>
     </attribute>
     <attribute name="horizontalHeaderHighlightSections">
      <bool>false</bool>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
     <column/>
    </widget>
   </item>
   <item row="5" column="0" colspan="7">
    <widget class="QTableView" name="STTable">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="horizontalScrollMode">
      <enum>QAbstractItemView::ScrollPerPixel</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>60</number>
     </attribute>
    </widget>
   </item>
   <item row="2" column="0" colspan="5">
    <widget class="QLabel" name="ExplicitSpikeTimesL">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>21</height>
      </size>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Spike Times</string>
     </property>
    </widget>
   </item>
  </layout>
  <zorder>widget</zorder>
  <zorder>buttonBox</zorder>
  <zorder>STTable</zorder>
  <zorder>SpikesL_2</zorder>
  <zorder>instTable</zorder>
  <zorder>label</zorder>
  <zorder>btnLoadST</zorder>
  <zorder>btnSaveST</zorder>
  <zorder>ExplicitSpikeTimesL</zorder>
  <zorder>Period</zorder>
  <zorder>PeriodU</zorder>
  <zorder>Loop</zorder>
  <zorder>label_2</zorder>
 </widget>
 <tabstops>
  <tabstop>SGLUTableCombo</tabstop>
  <tabstop>VSpike</tabstop>
  <tabstop>Width</tabstop>
  <tabstop>VRest</tabstop>
  <tabstop>BurstDetectionCombo</tabstop>
  <tabstop>tUnder</tabstop>
  <tabstop>tOver</tabstop>
  <tabstop>Loop</tabstop>
  <tabstop>Period</tabstop>
  <tabstop>btnLoadST</tabstop>
  <tabstop>btnSaveST</tabstop>
  <tabstop>STTable</tabstop>
  <tabstop>instTable</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SpikeGenDlg</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SpikeGenDlg</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
